cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/Datastore.java

```diff
@@ -338,5 +342,95 @@ void resolveConflictsForDocument(String docId, ConflictResolver resolver)
      */
     void compact();
 
+    /**
```

Comment:
Just to some up from our conversation, we're going to put the query methods on the Datastore interface, in order to make it more inline with CDTDatastore.